src/connectionController.ts

```diff
@@ -714,32 +698,21 @@ export default class ConnectionController {
       : '';
   }
 
-  isConnectionWithIdSaved(connectionId: string | null): boolean {
-    if (connectionId === null) {
-      return false;
-    }
-
-    return !!this._connections[connectionId];
-  }
-
-  getConnectingConnectionId(): string | null {
-    return this._connectingConnectionId;
+  getActiveDataService(): DataService | null {
+    return this._activeDataService;
   }
 
-  getConnectionStringFromConnectionId(connectionId: string): string {
-    return this._connections[connectionId].connectionModel.driverUrlWithSsh;
+  getMongoClientConnectionOptions(): { url: string; options: MongoClientOptions; } | undefined {
+    return this._activeDataService?.getMongoClientConnectionOptions();
   }
 
-  isCurrentlyConnected(): boolean {
-    return this._activeDataService !== null;
-  }
 
-  getActiveDataService(): null | DataServiceType {
-    return this._activeDataService;
+  getActiveDerivedConnectionModel(): ConnectionModel {
```

Comment:
Is this function used? If it's unused maybe we can remove it? I might just be missing where it's called.

Reply:
Leftover :)